modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/EndpointHelper.java

```diff
@@ -124,6 +125,34 @@ public Page<Map<String, Object>> getResponseEntityMapsPage(
 			responseEntityMaps, pagination, objectEntriesPage.getTotalCount());
 	}
 
+	public Map<String, Object> postObjectEntry(
+			long companyId, Map<String, Object> propertiesMap,
+			APIApplication.Schema requestSchema,
+			APIApplication.Schema responseSchema, String scopeKey)
+		throws Exception {
+
+		Map<String, Object> objectPropertiesMap = new HashMap<>();
+
+		for (APIApplication.Property property : requestSchema.getProperties()) {
+			Object value = propertiesMap.get(property.getName());
```

Comment:
Does it work for the complex attributes? (the related elements)

Reply:
No but because the scope of the story is only setting the values of the Main Object Definition. The related elements will be developed in another story.